src/main/scala/org/broadinstitute/dsde/agora/server/webservice/ApiService.scala

```diff
@@ -0,0 +1,111 @@
+package org.broadinstitute.dsde.agora.server.webservice
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.event.{Logging, LoggingAdapter}
+import akka.event.Logging.LogLevel
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.StatusCodes._
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server._
+import akka.http.scaladsl.server.RouteResult.Complete
+import akka.http.scaladsl.server.directives.{DebuggingDirectives, LogEntry, LoggingMagnet}
+import akka.stream.Materializer
+import akka.stream.scaladsl.Sink
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.agora.server.SwaggerRoutes
+import org.broadinstitute.dsde.agora.server.dataaccess.permissions.PermissionsDataSource
+import org.broadinstitute.dsde.agora.server.exceptions._
+import org.broadinstitute.dsde.agora.server.ga4gh.Ga4ghService
+import org.broadinstitute.dsde.agora.server.webservice.configurations.ConfigurationsService
+import org.broadinstitute.dsde.agora.server.webservice.methods.MethodsService
+import org.broadinstitute.dsde.agora.server.webservice.permissions.{EntityPermissionsService, MultiEntityPermissionsService, NamespacePermissionsService}
+import org.broadinstitute.dsde.workbench.model.{ErrorReport, WorkbenchException, WorkbenchExceptionWithErrorReport}
+import org.broadinstitute.dsde.agora.server.errorReportSource
+import spray.json.DefaultJsonProtocol
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ApiService extends LazyLogging with SprayJsonSupport with DefaultJsonProtocol {
+
+  // Required for marshalling errors:
+  import org.broadinstitute.dsde.agora.server.model.AgoraApiJsonSupport._
+  import org.broadinstitute.dsde.workbench.model.ErrorReportJsonSupport._
+
+  def handleExceptionsAndRejections: Directive[Unit] = handleExceptions(exceptionHandler) & handleRejections(rejectionHandler)
+
+  val exceptionHandler: ExceptionHandler = {
+    ExceptionHandler {
+      case withErrorReport: WorkbenchExceptionWithErrorReport =>
+        complete(withErrorReport.errorReport.statusCode.getOrElse(InternalServerError), withErrorReport.errorReport)
+      case workbenchException: WorkbenchException =>
+        val report = ErrorReport(Option(workbenchException.getMessage).getOrElse(""), Some(InternalServerError), Seq(), Seq(), Some(workbenchException.getClass))
+        complete(InternalServerError, report)
+      case e: IllegalArgumentException => complete(BadRequest, e)
```

Comment:
Why do we want to wrap them in `AgoraException`s?

Reply:
Some of the choices we made here were because that was how it was done in the original code and we tried to replicate that as much as possible. If wrapping doesn't introduce any test errors, I'm fine with it.